discretisedfield/mesh.py

```diff
@@ -396,6 +396,13 @@ def axis_points(self, axis, /):
         for i in range(self.n[axis]):
             yield self.index2point((0, 0, 0))[axis] + i*self.cell[axis]
 
+    def cell_points(self, axis, /):
```

Comment:
What do you think about making this `mesh.x`, `mesh.y`... similar to `field.x`?

Reply:
A different interface for this would be good. Also because `cell_points` and `axis_points` is very similar.

An additional slight difficulty is the fact that we now add support for `xarray`. `xarray` also supports getting coordinates (similar to `axis_points`) of the centre points of cells, e.g. `<xarray-field>.x`. So it could be confusing if `<mesh>.x` returns different points (and a different shape `n+1`)

Maybe we could have something like the following instead.
- Alternative 1 (no `mesh.x`)
  ```python
  >>> mesh = df.Mesh(...)
  >>> mesh.midpoints.x  # replaces axis_points
  <generator ...>
  >>> mesh.vertices('x')  # replaces cell_points
  <generator ...>
  ```
  As indicated `x` could either be passed as a string to the two methods or we could use e.g. `namedtuple`s and have properties.
- Alternative 2
  ```python
  >>> mesh = df.Mesh(...)
  >>> mesh.x
  <edge object>  # new object
  >>> mesh.x.vertices  # replaces cell_points
  <generator ...> 
  >>> mesh.x.midpoints  # replaces axis_points
  <generator ...>
  ```

Reply:
I like `vertices` vs. `midpoints` to explain the difference. The interface would be then, e.g. `mesh.vertices.x`, `mesh.midpoints.z`,... Namedtuple should be the best way forward here.